src/pages/patientView/vafPlot/VAFPlot.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from 'react';
+import $ from 'jquery';
+import {AlleleFreqPlotMulti} from './legacyVAFCode.js';
+
+export type IVAFPlotProps = {
+    data?: { [s: string]:number[] };
+    order?: { [s:string]:number };
+    colors?: { [s: string]:string };
+    labels?: { [s:string]:string };
+    width?: number;
+    height?: number;
+    margin_left?: number;
+    margin_right?: number;
+    margin_top?: number;
+    margin_bottom?: number;
+    xticks?: number;
+    yticks?: number;
+    label_font_size?: string;
+    nolegend?: boolean;
+    init_show_histogram?: boolean;
+    init_show_curve?: boolean;
+    show_controls?: boolean;
+};
+export class VAFPlot extends React.Component<IVAFPlotProps, {}> {
+
+    constructor(props) {
+        super(props)
+
+        // Compute contingent default params
+        let label_dist_to_axis = (this.props.xticks === 0) ? 13: 30;
+
+        let options = {
+            label_font_size: this.props.label_font_size,
+            xticks: this.props.xticks,
+            yticks: this.props.yticks,
+            nolegend: this.props.nolegend,
+            margin: {
+                top: this.props.margin_top,
+                left: this.props.margin_left,
+                right: this.props.margin_right,
+                bottom: this.props.margin_bottom
+            },
+            width: this.props.width,
+            height: this.props.height
+        };
+        // margin bottom must be computed before height because height computation makes use of it
+        options.margin.bottom = (isNaN(options.margin.bottom) ? (30 + (label_dist_to_axis / 2)) : options.margin.bottom);
+        options.height = isNaN(options.height) ? (((500 + label_dist_to_axis) / 2) - options.margin.top - options.margin.bottom) : options.height;
+        this.state = {
+            show_histogram: this.props.init_show_histogram,
+            show_curve: this.props.init_show_curve,
+            options: options
+        };
+
+        this.toggleShowHistogram = this.toggleShowHistogram.bind(this);
+        this.toggleShowCurve = this.toggleShowCurve.bind(this);
+    }
+
+    componentDidMount() {
+        AlleleFreqPlotMulti(this, this.props.data, this.state.options, this.props.order, this.props.colors, this.props.labels);
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+        // Hack around React: handle updating here, and return false
+        if (this.histogram_checkbox) {
```

Comment:
if we determine that rendering AlleleFreqPlotMulti is not prohibitively expensive, we could err on the side of just fully re-rendering it (i.e. in render method).  That way we can avoid these kind of shenanigans and be doing real React.   Basically my point is that we should actually be biased toward real React flow in cases where we're certain there isn't a performance issue and shouldComponentUpdate conditions are simple.  We can discuss.